bundles/org.openhab.binding.mecmeter/src/main/java/org/openhab/binding/mecmeter/MecMeterDeviceConfiguration.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mecmeter;
+
+/**
+ * The {@link MecMeterDeviceConfiguration} is the class used to match the
+ * thing configuration.
+ *
+ * @author Florian Pazour - Initial contribution
+ * @author Klaus Berger - Initial contribution
+ */
+public class MecMeterDeviceConfiguration {
+    public String ip = "";
+    public String username = "admin";
+    public String password = "12345";
+    public int refreshInterval = 5;
+
+    public MecMeterDeviceConfiguration() {
+        super();
+    }
+
+    public String getIp() {
+        return ip;
+    }
+
+    public void setIp(String inetaddress) {
+        ip = inetaddress;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String user) {
+        username = user;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String pw) {
+        password = pw;
+    }
+
+    public int getRefreshInterval() {
+        return refreshInterval;
+    }
+
+    public void setRefreshInterval(int ri) {
+        refreshInterval = ri;
+    }
+
+    public String isValid() {
+        if (ip == null || ip.isBlank()) {
+            return "Missing IP";
+        }
+        if (ip.contains("0.0.0.0")) {
+            return "IP not set";
+        }
+        if (username == null || username.isBlank()) {
+            return "username is missing";
+        }
+        if (password == null || password.isBlank()) {
```

Comment:
The password is required so it cannot be null.

Reply:
What happens if the user forgets to specify the parameter in textual configuration?

Reply:
@fwolter The Thing won't be initialized, if some mandatory parameter is missing.